src/lib.rs

```diff
@@ -1,16 +1,23 @@
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::collections::UnorderedMap;
-use near_sdk::{env, near_bindgen};
+use near_sdk::{env, near_bindgen, init, AccountId, Promise};
 
 mod models;
 use models::{Article, ArticleMeta, Rating, RatingAction, ONE_NEAR};
 
+#[cfg(test)]
+use models::ParsedReceipt;
+
 mod constants;
 use constants::ERR_ARTICLE_NOT_FOUND;
 
 // the upvote/downvote ratio, below which the article is hidden away (#12)
 const ARTICLE_VISIBILITY_VOTING_RATIO: f32 = 3.0 / 7.0;
 
+// sane constraints for donation amount
+const MIN_DONATION_AMOUNT: u128 = 1_000_000_000_000_000_000_000_000; // 1 $NEAR in yoctoNEAR
```

Comment:
Is there a mechanism to convert NEAR to yoctoNEAR?

I remember in the AssemblyScript SDK, there are built-in functions to convert between the two. Maybe we should make 2 functions in Rust too.

Reply:
what you referred to was the [near-api-js](https://docs.near.org/docs/api/naj-quick-reference#near--yoctonear). Unfortunately, the rust-lang `near-sdk` doesn't seem to have anything equivalent to that, so I added a quick util function to achieve sth similar.